manifests/cli.pp

```diff
@@ -0,0 +1,63 @@
+# = Pulpcore command line interface
+#
+# This class installs the Pulpcore command line interface (CLI).
+#
+# === Parameters:
+#
+# $pulpcore_url::       URL on which Pulpcore runs
+#
+# $username::           Username for authentication
+#
+# $password::           Password for authentication
+#
+# $cert::               Client certificate for authentication
+#
+# $key::                Client key for authentication
+#
+# === Advanced parameters:
+#
+# $api_root::           Absolute API base path on server (not including 'api/v3/')
+#
+# $verify_ssl::         Verify SSL connection
+#
+# $dry_run::            Trace commands without performing any unsafe HTTP calls
+#
+# $version::            pulp-cli package version, it's passed to ensure parameter of package resource
+#                       can be set to specific version number, 'latest', 'present' etc.
+#
+class pulpcore::cli (
+  Optional[Stdlib::HTTPUrl] $pulpcore_url = $pulpcore::cli::params::pulpcore_url,
+  String $version = $pulpcore::cli::params::version,
+  Optional[String] $username = $pulpcore::cli::params::username,
+  Optional[String] $password = $pulpcore::cli::params::password,
+  Optional[String] $cert = $pulpcore::cli::params::cert,
+  Optional[String] $key = $pulpcore::cli::params::key,
+  Optional[String] $api_root = $pulpcore::cli::params::api_root,
+  Boolean $verify_ssl = $pulpcore::cli::params::verify_ssl,
+  Boolean $dry_run = $pulpcore::cli::params::dry_run,
+) inherits pulpcore::cli::params {
+  package { 'pulp-cli':
+    ensure => $version,
+  }
+  -> file { '/etc/pulp/cli.toml':
+    ensure  => file,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
```

Comment:
There's a password in there. Should it really be world readable?

Reply:
That's one of the things I wanted to talk about. This file *can* have a PW in it, but in the Katello scenario it won't -- we use certs there, and those are properly protected already (only `root` and members of `foreman` can read those).

So I'd prefer to leave that file readable as that'll make it easier for non-root-but-foreman-members to use it.

But not in the case when there is a PW in it :see_no_evil: 

One possibility is just not to allow user/passwords at all (as we don't need them in Katello). The other is making the mode "smart" (but I don't like smart code).

Or, well, bite the bullet and make the file stricter owned.

Reply:
If I'm reading https://github.com/pulp/pulp-cli/blob/e7fe445e605bbcfc0a665ba1c53a42add719a191/pulpcore/cli/common/__init__.py#L49-L59 correctly, it does merge.

Maybe @mdellweg wants to chime in?

Reply:
Yes, that is the "level 1 merge". If you specify
```toml
[cli]
username = "bob"
password = "not my password"
```
in one and
```toml
[cli]
password = "most secret"
```
in the other, the `[cli]` sections will be combined by overwriting individual settings.